phantombuster/web-browser-extension
# fix: inject button in the proper placement

## 📝 What is this PR

[FE-106](https://phantombuster.atlassian.net/jira/software/projects/FE/boards/46?selectedIssue=FE-106)
> The web extension doesn’t inject the “Connect to service“ button in the proper placement. This is because the input DOM node selector changed after rebranding the setup page.
![image](https://user-images.githubusercontent.com/12436800/172897543-36b4b969-55fb-422f-8c69-d9ed91d3c7ae.png)

## ⚠️  Note
The web extension is working now for the deploy previews domain. We absolutely need that to test the extension properly.

## 🧪 How to test

### First, build and load the extension into your browser

- run `nvm use` on the parent folder of the web extension
- run `npm i`
- run `npm run build`
- Follow [these steps](https://developer.chrome.com/docs/extensions/mv3/getstarted/#unpacked) to load an unpacked extension into chrome.
- Follow [these steps](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Your_first_WebExtension#installing) to load the extension into firefox.

### Testing the extension
- Check the new icon of the extension
- Go to a deploy preview (rebranded website), use different scripts (twitter, facebook, linkedin ...), and check the new styling of the `Connect to {service}` button.
- Check that the extension is still working properly.


![twitter extension](https://user-images.githubusercontent.com/12436800/172898793-ca4450f1-f09c-4f56-981a-31e4255accfc.gif)
